backend/routes/api.php

```diff
@@ -14,10 +15,16 @@
 
 Route::post('/rooms/{room}/roomcode', [RoomController::class, 'generateRoomCode'])->middleware('auth:sanctum');
 
+Route::get('/rooms/{room}/details', [RoomController::class, 'showDetails']);
```

Comment:
this route returns the same information, no, even less information than `/rooms/{room}` (defined in line 14 with show) but is called details? i don't think it is necessary.
you could also explore the possibility of also sending the required data when requesting team which would eliminate the need for an extra request